libs/designer/src/lib/core/utils/openapi/inputsbuilder.ts

```diff
@@ -1,7 +1,7 @@
 import type { InputParameter } from '@microsoft/parsers-logic-apps';
 
 export interface OpenApiConnectionSerializedInputs {
-  parameters: Record<string, unknown>;
+  parameters?: Record<string, unknown>;
 }
 
 export class OpenApiOperationInputsBuilder {
```

Comment:
Why is there a need for this inputsBuilder? It could have been done using the parameter.alias and the manifest deserializer.

Reply:
That wasn't brought up in #1721 but the logic can be moved there instead.

For now, I'd like to get this fix in so we don't have to deal with this error, and we can refactor the logic as needed after.

Reply:
inputsLocation is not correctly added in your manifest, this should ne easily handled without a inputsbuilder, we cannot keep doing so many cleanups after GA... please fix it the right way...
make sure your openApiConnection manifests have [inputs, parameters] in the inputsLocation and then fix the code which serializes or deserializes if there are bugs.

Reply:
The problem is that the code you're asking us to remove is already in, and if we didn't have this fix, the cleanup would be after GA anyway. This is a fix for that code. We risk breaking our scenarios at Build if we try to do some funky refactoring now.

Per your specific concern, yes, `inputsLocation` is correct:

![image](https://user-images.githubusercontent.com/1350074/235727112-8dd43ebf-c6f6-4303-aae9-bc5c8736df47.png)

The problem is that the object specified by that path can be `undefined`. I will look at the existing code for how we can leverage it, but effectively what you're asking us to do is jeopardize the functionality of OpenAPI before Build.

Reply:
@preetriti1 Okay, this would be the alternative change. We'd remove the inputs builder, and do this instead:

![image](https://user-images.githubusercontent.com/1350074/235750735-29edb6f1-526c-4bda-abfb-c46050746342.png)

![image](https://user-images.githubusercontent.com/1350074/235750830-ea89f729-44b4-4f90-9988-fd1d7bd5479f.png)

I still feel this is too high-risk to push in on such a tight timeline. However, it does seem to resolve the issue fixed in #1721 and the issue in this current PR, and allows us to get rid of the inputs builder.

Reply:
Updates have been made.